Content.Server/StationEvents/Events/SolarFlare.cs

```diff
@@ -0,0 +1,86 @@
+using Content.Server.GameTicking.Rules.Configurations;
+using Content.Server.Radio.Components;
+using Content.Server.Radio;
+using Robust.Shared.Random;
+using Content.Server.Light.EntitySystems;
+using Content.Server.Light.Components;
+
+namespace Content.Server.StationEvents.Events
+{
+    public sealed class SolarFlare : StationEventSystem
+    {
+        [Dependency] private readonly PoweredLightSystem _poweredLight = default!;
+
+        public override string Prototype => "SolarFlare";
+
+        private bool _onlyJamHeadsets = true;
+        private HashSet<string> _affectedChannels = new();
+        private float _endAfter = 0.0f;
+        private bool _running = false;
+        private float _lightBurnChance = 0.0f;
+        private float _lightChangeColorChance = 0.0f;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<ActiveRadioComponent, RadioReceiveAttemptEvent>(OnRadioSendAttempt);
+        }
+
+        public override void Added()
+        {
+            if (Configuration is not SolarFlareEventRuleConfiguration ev)
+                return;
+            base.Added();
+            _onlyJamHeadsets = ev.OnlyJamHeadsets;
+            _endAfter = RobustRandom.Next(ev.MinEndAfter, ev.MaxEndAfter);
+            _affectedChannels = ev.AffectedChannels;
+            _lightBurnChance = ev.LightBreakChance;
+        }
```

Comment:
store a private SolarFlareEventRuleConfiguration field and just cast once inside of Added instead of copypasting all the data, something like 
```cs
    private SolarFlareEventRuleConfiguration _event = default!;

	private  override void Initialize() ....;

    public override void Added()
    {
        base.Added();

        var ev = (SolarFlareEventRuleConfiguration) Configuration;
        _event = ev;
        _event.EndAfter = RobustRandom.Next(ev.MinEndAfter, ev.MaxEndAfter);
    }
```